### ISSUE TITLE
Suche nach Wertpapier: No default Proxy found

### ISSUE THREAD

**Issue by GralfR** [littleyoda/hibiscus.depotviewer#96]
Hi!
Versuche in Hibiscus.depotviewer manuell ein Wertpapier hinzuzufügen. Im Grunde möchte ich nur ein Offline Depot führen, um eine Liste mit Wertpapieren und aktuellem Kurs anstelle meines nicht per HBCI abrufbaren Depots zu haben.
Hatte es so verstanden, daß ich Unter > Wertpapiere > Hinzufügen entsprechende ISIN/WKN einfügen kann und dann ähnlich einem online synchronisierten Konto meine Kurse abfragen kann.

Leider scheitere ich bereits an dem Hinzufügen eines Wertpapieres. Gebe ich unter > hinzufügen bei "nach einem Wertpapier suchen" eine ISIN oder einen Namen ein und klick auf Suchen, passiert nix. Es werden keine Wertpapiere angezeigt. Als Suchbegriff habe ich z.B. eine ISIN direkt per Copy/Paste aus einer Webseite kopiert, oder diverse Phrasen wie axa oder vw versucht.

Im Log gibt es drei neue Einträge nach dem Klick auf Suchen:
```
Using system proxy settings: [DIRECT]
No default Proxy found
Keine gültige Proxy-Einstellunge gefunden. (true)
```
In den Jameica Einstellungen habe ich unter Proxy "Systemeinstellungen verwenden" angeklickt. Im Grunde aber auch egal, denn ich habe keinen Proxy. Alles andere funktioniert soweit (Hibiscus Bankkonten synchronisieren, Surfen über Browser).

**Comment by GralfR** [comment]
Habe jetzt mal testweise einen systemweiten Proxy konfiguriert. Wenn ich nun im Depotviewer unter Wertpapiere auf Suchen klicke kommen folgende Meldungen im Log, aber kein Suchergebnis:
```
Using system proxy settings: [HTTP @ 173.249.5.7:8080]
Setting Proxy to com.gargoylesoftware.htmlunit.ProxyConfig@514c6ca0
```
Egal welchen Proxy ich konfiguriere, immer kommt "Setting Proxy to com.gargoylesoftware.htmlunit.ProxyConfig@". Es ändern sich lediglich die Zeichen hinter dem @.
Es sieht so aus, als ob die Meldungen aus [diesem Modul](https://github.com/littleyoda/hibiscus.depotviewer/blob/master/src/de/open4me/depot/abruf/utils/HtmlUtils.java) stammen.
Das bringt mich aber auch nicht weiter. Kann sich das bitte mal jemand ansehen und auf dem eigenen System probieren, ob die Wertpapiersuche funktioniert? Mittlerweile glaube ich an einen Bug. So viel kann man doch als User nicht falsch machen, wenn man keinen Proxy verwendet...

**Comment by faiteanu** [comment]
Hast du vorher versucht, Daten über das Menü Depot-Viewer -> "Download 'Gesamtliste aller auf Xetra handelbaren Wertpapiere' " zu laden? Bei Yahoo Finance hat sich eventuell die Seitenstruktur geändert, so dass die Daten von dort nicht mehr richtig geladen werden können.

**Comment by GralfR** [comment]
Danke für den Hinweis. Habe ich jetzt mal versucht.
Der Download wird ohne Fehlermeldungen abgeschlossen - egal ob mit oder ohne Proxy.
Von den im Menüpunkt erwähnten 35MB ist aber keine Spur. Im Jameica-Verzeichnis wird eine CSV-Datei angelegt, die aber nur 1kB groß ist und nur die Headline enthält.
Wenn ich die im Log angegebene URL zur csv-Datei direkt im Browser eingebe, wird die Datei zwar geladen, ist aber auch nur 3kB groß.
Falls bei Yahoo oder xetra oder wo auch immer was geändert wurde, wird das ja wohl ein stetiger Update-Kampf bleiben, oder? Fehlt dann nicht im Depot-Viewer ein Workaround? In meinem Fall möchte ich z.B. nur manuell Wertpapiere zu einem Offline-Depot hinzufügen und von Zeit zu Zeit den Kurs aktualisieren. Ich habe kein Problem damit, die nötigen Daten manuell einzugeben. Wenn es also einen Dialog gäbe, wo ich im Notfall, wenn die ganze Automation mal wieder nicht funktioniert, alle nötigen Daten manuell eingeben kann, reicht das zumindest zur Überbrückung, bis entsprechende URLs wieder korrigiert sind. Für die paar eigenen Wertpapiere kann ich mir WKN, ISIN, aktuellen Kurs und was sonst noch nötig ist, manuell raussuchen, wenn das vielleicht auch noch gut dokumentiert ist. So stimmt dann zumindest der Gesamtsaldo wieder. Historische Werte muß man sich dann eben erstmal anders besorgen.

Nutzt denn sonst niemand diese Funktion? Versuche gerade erst auf Hibiscus umzusteigen. Komme von StarMoney, aber das nervt gewaltig.

**Comment by littleyoda** [comment]
[Leider habe ich gerade nicht so viel Zeit für das Plugin, wie eigentlich nötig wäre]

Wahrscheinlich hat sich die URL der Datei wieder mal geändert.

Die Kursaktualisierung war und ist immer ein Kampf. Deshalb hatte ich dafür eine andere Library genutzt, die aber leider nicht mehr gepflegt wird.  faiteanu hatte die Library mal angepasst. Aber ich habe seine Version noch nicht eingebaut.

**Comment by littleyoda** [comment]
Mit der nächsten Version (spätestens morgen abend) wird das Herunterladen und Suchen in der Xetra-Liste wieder funktionieren.
Könntet du es dann nochmal testen, ob du es auch mit Proxy herunterladen kannst?

**Comment by GralfR** [comment]
Ja, mache ich. Aber eigentlich brauche ich gar keinen Proxy-Support. Ich habe eine Direkt-Verbindung, die nicht funktioniert. Wegen der Fehlermeldungen habe ich nur versucht, ob es mit Proxy geht.
Wichtig ist, daß man manuelle Wertpapiere einfügen kann. Wie gesagt, wenn's automatisch nicht funktioniert, dann halt manuell alles in einem Dialog eingeben. Reicht vollkommen. So oft macht man das ja nicht.

**Comment by GralfR** [comment]
Habe gerade auf 0.6 upgedatet und getestet: Suche funktioniert - mit und ohne Proxy. Auch der Download der Gesamtliste bringt nun eine CSV-Datei mit Inhalt - wobei mit 256kB weit entfernt von den 35MB, die im Menüpunkt angegeben sind.
Ohne Proxy bleiben aber weiterhin die Fehlermeldungen aus meinem ersten Post. Macht jetzt noch weniger Sinn: man nutzt die systemweiten Einstellungen, in denen kein Proxy hinterlegt ist, die Verbindung kommt zustande, und trotzdem gibt es eine ERROR-Meldung, daß kein Proxy gefunden wurde. Widerspricht sich irgendwie.

Auch blöd ist, daß nicht alle ISIN/WKN gefunden werden. Wenn ich z.B. in die Suche "gold" eingebe, wird kein Feingold als Rohstoff (WKN: 965515  ISIN: XC0009655157) gefunden, so daß ich das nicht hinzufügen kann. Liegt vermutlich daran, daß es kein Wertpapier ist.
Habe dann einen kleinen Workaround gemacht: 
1. irgendein gefundenes Wertpapier hinzufügen (habe das XETRA-GOLD genommen).
2. danach unter Wertpapiere das neue Wertpapier bearbeiten und die gewünschten Daten (neuer Name, WKN, ISIN) eingeben
3. fertig. Das Wertpapier ist drin und wird mit dem richtigen Kurs aktualisiert.

Genau diese Prozedur wünsche ich mir: Einfach manuell ein Wertpapier ohne vorige Suche hinzufügen können. Wie der Dialog zum Bearbeiten zeigt, braucht man anscheinend nicht mehr als Name, WKN, ISIN. Das kann man ganz einfach von jeder Finanz-Seite erfahren. Natürlich ist die Suche ganz komfortabel, aber wenn sie nicht funktioniert, ist man aufgeschmissen und alle haben Streß für mehrere Tage, wie dieser Thread beweist. Es fehlt einfach der Bearbeiten-Dialog als manuelles Hinzufügen.
